### DISCUSSION TITLE
Implementing a Bundler after-install-all hook

### DISCUSSION THREAD

**Discussion by matanskii** [ruby/rubygems#7880] [Q&A]
Hi,
going over the documentation here https://bundler.io/guides/bundler_plugins.html#using-bundler-hooks - I didn't understand where to actually put this part:
```
      Bundler::Plugin.add_hook(Bundler::Plugin::Events::GEM_AFTER_INSTALL) do |dependencies|
        # Do something with the dependencies
        puts "======after all====="
      end
```
Does it go into the my_plugin.rb file? Under any specific module, like the command override? I can't get it to work, please help.
Thank you!

**Comment by jeromedalbert** [comment]
Looking at a few [example plugins](https://bundler.io/guides/plugins.html) it looks like a lot of them add their hooks in `lib/my_plugin.rb`, for example:

- https://github.com/sergey-alekseev/bundler-licensed/blob/8b8967916313e4a36fc6edbecb893b38e374b8a7/lib/bundler/licensed.rb#L9
- https://github.com/flavorjones/bundler-as_of/blob/0a19cfa24faf954633a8072ab97e60163c8fc5c4/lib/bundler/as_of.rb#L139
- https://github.com/jdar/bundler-changelogs/blob/95d279d028ff6a378db79d4cdba3792c3ea18cb7/lib/bundler/changelogs.rb#L20

Although this one seems to add their hook directly in the root `plugins.rb`: https://github.com/okuramasafumi/bundler-ctags_generator/blob/4a375def2f6aa769fd5e41ec7998b25f0b4cd2ad/plugins.rb#L3.

So it looks like any code inside or `require`d by the root `plugins.rb` file would work.
